src/main/kotlin/uk/gov/justice/digital/hmpps/incidentreporting/service/ReportService.kt

```diff
@@ -182,4 +185,64 @@ class ReportService(
       }
     }.getOrNull()
   }
+
+  @Transactional
+  fun changeReportStatus(id: UUID, changeStatusRequest: ChangeStatusRequest): Pair<ReportWithDetails, Boolean>? {
```

Comment:
it's a bit unpleasant returning a “pair” or values. the report might or might not have been updated and this is reflected by the boolean flag…

we can make something a bit like a tagged union?

Reply:
This returns the `ReportWithDetails` when the report is changing.

So I think returning `ReportWithDetails?` maybe be simpler and good enough.

e.g. returns a `ReportWithDetails` if the status changed, returns `null` when the status didn't change.

Reply:
I see, it's because you want this to also check if the report exists?

The client code could either pass the Report or you could just throw a "Report not found exception" in here (which I think it makes sense as this is asked to update a report and if it doesn't exist it's an invalid operation)